implementations/rust/ockam/ockam_api/src/cli_state/nodes.rs

```diff
@@ -223,55 +233,61 @@ impl CliState {
         pid: u32,
         signal: signal::Signal,
     ) -> Result<()> {
+        debug!(%pid, %signal, "sending kill signals to node's process");
         let pid = nix::unistd::Pid::from_raw(pid as i32);
-        signal::kill(pid, signal)
-            .or_else(|e| {
-                if e == Errno::ESRCH {
-                    tracing::warn!(node = %node_name, %pid, "No such process");
-                    Ok(())
-                } else {
-                    Err(e)
-                }
-            })
-            .map_err(|e| {
-                CliStateError::Io(std::io::Error::new(
-                    std::io::ErrorKind::Other,
-                    format!("failed to stop PID `{pid}` with error `{e}`"),
-                ))
-            })?;
-        debug!(name = %node_name, %pid, %signal, "sent stop signal to node's process");
+        let _ = self.send_kill_signal(pid, signal);
 
         // Wait until the node has fully stopped
         let timeout = std::time::Duration::from_millis(100);
-        let max_attempts = std::time::Duration::from_secs(5).as_millis() / timeout.as_millis();
+        let max_attempts = std::time::Duration::from_secs(3).as_millis() / timeout.as_millis();
         let mut attempts = 0;
-        let mut sys = System::new();
-        let pid = Pid::from_u32(pid.as_raw() as u32);
         loop {
-            sys.refresh_processes(ProcessesToUpdate::All);
-            if sys.process(pid).is_none() {
-                info!(name = %node_name, %pid, "node process exited");
-                self.notify_progress_finish_and_clear();
-                return Ok(());
+            match self.send_kill_signal(pid, signal) {
```

Comment:
Is there a point to send multiple the same signal multiple times? @davide-baldo told me there is no, but I haven't checked :)

Reply:
the first time it always returns OK, it just ack's that the signal was sent. You need to keep sending it until it returns the `Errno::ESRCH`, which signals that the process is effectively gone.